src/components/Quotas/QuotaEditor/QuotaForm.tsx

```diff
@@ -0,0 +1,107 @@
+import { SkeletonLoader } from '../../../helpers/SkeletonLoader';
+import { useNavigation } from '../../../hooks/useNavigation';
+import { useQuotaFormValues } from '../../../hooks/useQuotaFormValues';
+import useSaveQuota from '../../../hooks/useSaveQuota';
+import { useTranslation } from '../../../hooks/useTranslation';
+import { QuotaFormValues } from '../../../types/quotaFormTypes';
+import { Quota } from '../../../types/resourceTypes';
+import { getErrorMessage } from '../../../utils/utils';
+import QuotaFormFields from './QuotaFormFields';
+import useQuotasValidationSchema from './quotasValidationSchema';
+
+const QuotaFormToolbar = ({
+  onCancel,
+  handleSubmit,
+  isSubmitting,
+  isCreateFlow,
+}: {
+  onCancel: () => void;
+  handleSubmit: () => void;
+  isSubmitting: boolean;
+  isCreateFlow: boolean;
+}) => {
+  const { t } = useTranslation();
+  return (
+    <ActionGroup>
+      <Button
+        key="confirm"
+        variant="primary"
+        isLoading={isSubmitting}
+        name="confirm"
+        onClick={handleSubmit}
+        type="submit"
+      >
+        {isCreateFlow ? t('Create') : t('Save')}
+      </Button>
+
+      <Button key="cancel" onClick={onCancel} variant="link">
+        {t('Cancel')}
+      </Button>
+    </ActionGroup>
+  );
+};
+
+const QuotaForm = ({ quota }: { quota?: Quota }) => {
+  const { t } = useTranslation();
+  const [initialValues, initialValuesLoaded, initialValuesError] = useQuotaFormValues(quota);
+  const [validationSchema, validationSchemaLoaded, validationSchemaError] =
+    useQuotasValidationSchema(!quota);
+  const [save, saveLoaded] = useSaveQuota(quota);
+  const [submitError, setSubmitError] = React.useState<unknown>();
+  const history = useHistory();
+  const navigation = useNavigation();
+  const onSubmit = async (values: QuotaFormValues) => {
+    try {
+      await save(values);
+      navigation.goToClusterTemplatesPage('quotas');
+    } catch (err) {
+      setSubmitError(err);
+    }
+  };
+  return (
+    <AlertsContextProvider>
+      <SkeletonLoader
+        numRows={8}
+        loaded={saveLoaded && initialValuesLoaded && validationSchemaLoaded}
+        error={initialValuesError || validationSchemaError}
+      >
+        {initialValues && (
+          <Formik<QuotaFormValues>
+            initialValues={initialValues}
+            onSubmit={onSubmit}
+            validationSchema={validationSchema}
+          >
+            {({ isSubmitting, handleSubmit }) => (
+              <Stack hasGutter>
+                <Form isWidthLimited>
+                  <QuotaFormFields originalQuota={quota} />
+                </Form>
+                {submitError && (
+                  <Alert title={t('Failed to save the quota')} variant="danger" isInline>
+                    {getErrorMessage(submitError)}
+                  </Alert>
+                )}
+                <QuotaFormToolbar
+                  onCancel={() => history.goBack()}
```

Comment:
I'm getting

Avoid referencing unbound methods which may cause unintentional scoping of `this`.
If your function does not access `this`, you can annotate it with `this: void`, or consider using an arrow function instead

It might be that history.goBack uses "this", any idea how to resolve this?